pkg/providers/tinkerbell/upgrade.go

```diff
@@ -292,6 +307,41 @@ func (p *Provider) ValidateNewSpec(ctx context.Context, cluster *types.Cluster,
 	return nil
 }
 
+func validateControlPlaneMachineConfigRefUnchanged(current, desired *v1alpha1.Cluster) error {
```

Comment:
I appreciate that this pattern of naming already exists, but it feels way too long. Thoughts on a shorter name? We could add a function code comment to help clarify it, if needed. Maybe something like,`func validateRef` or `func validateCPRef`, etc?

Also, could this function just take in the `v1alpha1.Ref` struct instead of the whole `v1alpha1.Cluster`?

Reply:
I honestly just prefer the clarity in this instance. If its not part of the function name, its part of a comment on the function that someone has to read anyway.

If we drop this to `validateCPRef(v1alpha1Ref)` then I'll just get rid of the function all together because it'll be superfluous.

Reply:
I just got rid of this func, it felt superfluous.